test/cpp_extensions/cuda_extension.cu

```diff
@@ -0,0 +1,29 @@
+// NOTE: This is a copy of cuda_extension_kernel.cu. It's kept here to test
+// collision handling when a C++ file and CUDA file share the same filename.
+// Setuptools can't deal with this at all, so the setup.py-based test uses
```

Comment:
It's a pity that we won't be able to have setuptools also accept files with the same name but different extensions.

I suppose then that in this case the recommended way of creating extensions will be to have different names for the files, as I suppose it will be good practice to write extensions that work both using the setuptools as the jit compilation.